src/main/java/buchen/rijks/RijksSearchFrame.java

```diff
@@ -0,0 +1,102 @@
+package buchen.rijks;
+
+import com.andrewoid.ApiKey;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+import java.net.URL;
+
+public class RijksSearchFrame extends JFrame {
+    private final JTextField searchField;
+    private final JPanel imagePanel;
+    private int page = 0;
+    RijksService service = new RijksServiceFactory().create();
+
+    public RijksSearchFrame() {
+        setSize(700, 800);
+        setTitle("Rijks");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        setLayout(new BorderLayout());
+        searchField = new JTextField();
+        searchField.setText("Rembrandt van Rijn");
+
+        JButton prevButton = new JButton("Previous");
+        JButton nextButton = new JButton("Next");
+
+        nextButton.addActionListener(e -> {
+            page++;
+            update();
+        });
+        prevButton.addActionListener(e -> {
+            if (page > 0) {
+                page--;
+                update();
+            }
+        });
+        searchField.getDocument().addDocumentListener(
+                (SimpleDocumentListener) e -> {
+                    page = 0;
+                    update();
+                });
+
+        JPanel searchPanel = new JPanel(new BorderLayout());
+        searchPanel.add(searchField, BorderLayout.CENTER);
+        searchPanel.add(prevButton, BorderLayout.WEST);
+        searchPanel.add(nextButton, BorderLayout.EAST);
+
+        imagePanel = new JPanel(new GridLayout(0, 5));
+        add(searchPanel, BorderLayout.NORTH);
+        add(imagePanel, BorderLayout.CENTER);
+
+        update();
+    }
+
+    private void update() {
+        SwingUtilities.invokeLater(() -> {
+            ApiKey apiKey = new ApiKey();
+
+            ArtObjects collection;
+            if (!searchField.getText().isEmpty()) {
+                collection = service.query(apiKey.get(), page, searchField.getText()).blockingGet();
```

Comment:
blockingGet() is only allowed in tests. 
You need to use subscribe in applications.